### PR TITLE
feat: primitive tests 🧪

### LINKED ISSUE #298: feat(test): primitives methods testing 🧪
# Feature Request

**Describe the Feature Request**

Right now, we rely on pallet's tests to catch problems in `Wrapper` types methods implementations. 
Instead, they should have their own unit tests.

**Acceptance Criteria**

- PR should increase the coverage of the repo

We should have unit tests for the following methods : 
- [x]  `CallEntryPointWrapper.execute`
- [x]  `EventWrapper` builder pattern
- [x] `Transaction.verify_tx_version`
- [ ] `Transaction.execute`
- [x] `Header.hash`

### PR DESCRIPTION
**Pull request by EvolveArt**
# Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [x] Testing
- [ ] Other (please describe):

# What is the current behavior?

Issue Number: Closes #298

### CODE REVIEW

File: crates/primitives/starknet/src/execution/mod.rs [hunk 1]
```diff
@@ -9,10 +9,14 @@ use serde::de::Error as DeserializationError;
 use serde::{Deserialize, Deserializer, Serializer};
 use sp_core::Get;
 
-mod call_entrypoint_wrapper;
-mod contract_class_wrapper;
-mod entrypoint_wrapper;
-mod program_wrapper;
+/// Call Entrypoint Wrapper type
```
Comment (LucasLvy) [inline crates/primitives/starknet/src/execution/mod.rs:12]:
```suggestion
/// Call Entrypoint Wrapper related types
```

File: crates/primitives/starknet/src/execution/mod.rs [hunk 2]
```diff
@@ -9,10 +9,14 @@ use serde::de::Error as DeserializationError;
...
+pub mod call_entrypoint_wrapper;
+/// Contract Class Wrapper type
```
Comment (LucasLvy) [inline crates/primitives/starknet/src/execution/mod.rs:14]:
```suggestion
/// Contract Class Wrapper related types
```

File: crates/primitives/starknet/src/execution/mod.rs [hunk 3]
```diff
@@ -9,10 +9,14 @@ use serde::de::Error as DeserializationError;
...
+pub mod contract_class_wrapper;
+/// Entrypoint Wrapper type
```
Comment (LucasLvy) [inline crates/primitives/starknet/src/execution/mod.rs:16]:
```suggestion
/// Entrypoint Wrapper related types
```

File: crates/primitives/starknet/src/execution/mod.rs [hunk 4]
```diff
@@ -9,10 +9,14 @@ use serde::de::Error as DeserializationError;
...
+pub mod entrypoint_wrapper;
+/// Program Wrapper type
```
Comment (LucasLvy) [inline crates/primitives/starknet/src/execution/mod.rs:18]:
```suggestion
/// Program Wrapper typerelated types```

File: crates/primitives/starknet/src/tests/execution.rs [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+        calldata,
+        address,
+        ContractAddressWrapper::default(),
+    );
+
+    let block = Block::create_for_testing();
+
+    let error = entrypoint.execute(&mut test_state, block, [0; 32]).unwrap_err();
+
+    println!("{:?}", error);
```
Comment (LucasLvy) [inline crates/primitives/starknet/src/tests/execution.rs:56]: print ?

File: crates/primitives/starknet/src/tests/transaction.rs [hunk 6]
```diff
@@ -134,3 +138,58 @@ fn test_try_into_transaction_receipt_wrapper_with_too_many_events() {
...
+fn test_event_wrapper_new() {
+    let keys: BoundedVec<H256, MaxArraySize> = bounded_vec![H256::zero(), H256::from([1; 32])];
+    let data: BoundedVec<H256, MaxArraySize> = bounded_vec![H256::from([1; 32]), H256::from([2; 32])];
+    let from_address = ContractAddressWrapper::from(H256::from([3; 32]));
+
+    let event_wrapper = EventWrapper::new(keys.clone(), data.clone(), from_address);
+
+    assert_eq!(event_wrapper.keys, keys);
+    assert_eq!(event_wrapper.data, data);
+    assert_eq!(event_wrapper.from_address, from_address);
```
Comment (LucasLvy) [inline crates/primitives/starknet/src/tests/transaction.rs:152]: why not assert the full object with pretty assertions ?

File: crates/primitives/starknet/src/tests/transaction.rs [hunk 7]
```diff
@@ -134,3 +138,58 @@ fn test_try_into_transaction_receipt_wrapper_with_too_many_events() {
...
     assert_eq!(result.unwrap_err(), EventError::TooManyEvents);
 }
+
+#[test]
+fn test_event_wrapper_new() {
+    let keys: BoundedVec<H256, MaxArraySize> = bounded_vec![H256::zero(), H256::from([1; 32])];
+    let data: BoundedVec<H256, MaxArraySize> = bounded_vec![H256::from([1; 32]), H256::from([2; 32])];
+    let from_address = ContractAddressWrapper::from(H256::from([3; 32]));
+
+    let event_wrapper = EventWrapper::new(keys.clone(), data.clone(), from_address);
```
Comment (LucasLvy) [inline crates/primitives/starknet/src/tests/transaction.rs:148]: clone really needed ?

File: crates/primitives/starknet/src/tests/transaction.rs [hunk 8]
```diff
@@ -134,3 +138,58 @@ fn test_try_into_transaction_receipt_wrapper_with_too_many_events() {
...
+    assert_eq!(event_wrapper.data, data);
+    assert_eq!(event_wrapper.from_address, from_address);
+}
+
+#[test]
+fn test_event_wrapper_empty() {
+    let event_wrapper = EventWrapper::empty();
+    assert_eq!(event_wrapper.keys.len(), 0);
+    assert_eq!(event_wrapper.data.len(), 0);
+    assert_eq!(event_wrapper.from_address, ContractAddressWrapper::default());
```
Comment (LucasLvy) [inline crates/primitives/starknet/src/tests/transaction.rs:160]: why not full object ?

File: crates/primitives/starknet/src/tests/transaction.rs [hunk 9]
```diff
@@ -134,3 +138,58 @@ fn test_try_into_transaction_receipt_wrapper_with_too_many_events() {
...
+    let event_wrapper = EventWrapper::builder()
+        .with_keys(keys.clone())
+        .with_data(data.clone())
+        .with_from_address(ContractAddress::try_from(StarkFelt::new(from_address).unwrap()).unwrap())
+        .build()
+        .unwrap();
+
+    assert_eq!(event_wrapper.keys, keys);
+    assert_eq!(event_wrapper.data, data);
+    assert_eq!(event_wrapper.from_address, from_address);
```
Comment (LucasLvy) [inline crates/primitives/starknet/src/tests/transaction.rs:178]: same

File: crates/primitives/starknet/src/tests/transaction.rs [hunk 10]
```diff
@@ -134,3 +138,58 @@ fn test_try_into_transaction_receipt_wrapper_with_too_many_events() {
...
+        data: EventData(vec![StarkFelt::new([1; 32]).unwrap(), StarkFelt::new([2; 32]).unwrap()]),
+    };
+
+    let event_wrapper = EventWrapper::builder().with_event_content(event_content).build().unwrap();
+
+    let bounded_keys: BoundedVec<H256, MaxArraySize> = bounded_vec!(H256::zero());
+    let bounded_data: BoundedVec<H256, MaxArraySize> = bounded_vec![H256::from([1; 32]), H256::from([2; 32])];
+
+    assert_eq!(event_wrapper.keys, bounded_keys);
+    assert_eq!(event_wrapper.data, bounded_data);
```
Comment (LucasLvy) [inline crates/primitives/starknet/src/tests/transaction.rs:194]: same

File: crates/primitives/starknet/src/tests/utils.rs [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+            ClassHash(stark_felt!(SECURITY_TEST_CLASS_HASH)),
+        ),
+    ]);
+
+    CachedState::new(DictStateReader { class_hash_to_class, address_to_class_hash, ..Default::default() })
+}
+
+pub fn get_contract_class(contract_path: &str) -> ContractClass {
+    let path: PathBuf = [contract_path].iter().collect();
+    println!("path: {:?}", path);
```
Comment (LucasLvy) [inline crates/primitives/starknet/src/tests/utils.rs:62]: print ?

**Review by LucasLvy — CHANGES_REQUESTED** [review]

File: crates/primitives/starknet/src/tests/transaction.rs (hunk 7 above)
Comment (EvolveArt) [reply crates/primitives/starknet/src/tests/transaction.rs:148]: yes

File: crates/primitives/starknet/src/tests/execution.rs [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+        EntryPointTypeWrapper::External,
+        Some(selector),
+        calldata,
+        address,
+        ContractAddressWrapper::default(),
+    );
+
+    let block = Block::create_for_testing();
+
+    assert!(entrypoint.execute(&mut test_state, block, [0; 32]).is_err());
```
Comment (LucasLvy) [inline crates/primitives/starknet/src/tests/execution.rs:54]: use `assert_err`

File: crates/primitives/starknet/src/tests/transaction.rs [hunk 13]
```diff
@@ -134,3 +138,65 @@ fn test_try_into_transaction_receipt_wrapper_with_too_many_events() {
...
+        .build()
+        .unwrap();
+
+    pretty_assertions::assert_eq!(
+        event_wrapper,
+        EventWrapper {
+            keys: BoundedVec::<H256, MaxArraySize>::try_from(keys).unwrap(),
+            data: BoundedVec::<H256, MaxArraySize>::try_from(data).unwrap(),
+            from_address
+        }
```
Comment (LucasLvy) [inline crates/primitives/starknet/src/tests/transaction.rs:182]: same

File: crates/primitives/starknet/src/tests/transaction.rs [hunk 14]
```diff
@@ -134,3 +138,65 @@ fn test_try_into_transaction_receipt_wrapper_with_too_many_events() {
...
+    };
+
+    let event_wrapper = EventWrapper::builder().with_event_content(event_content).build().unwrap();
+
+    let bounded_keys: BoundedVec<H256, MaxArraySize> = bounded_vec!(H256::zero());
+    let bounded_data: BoundedVec<H256, MaxArraySize> = bounded_vec![H256::from([1; 32]), H256::from([2; 32])];
+
+    pretty_assertions::assert_eq!(
+        event_wrapper,
+        EventWrapper { keys: bounded_keys, data: bounded_data, from_address: ContractAddressWrapper::default() }
```
Comment (LucasLvy) [inline crates/primitives/starknet/src/tests/transaction.rs:200]: same

**Review by LucasLvy — APPROVED** [review]
lgtm but abdel is cringe

**Pull request merged**